File: src/main/java/net/moltendorf/Bukkit/LinkedWhitelist/storage/AbstractStorage.java
package net.moltendorf.Bukkit.LinkedWhitelist.storage;

import java.util.UUID;

/**
 * Created by moltendorf on 15/07/27.
 *
 * @author moltendorf
 */
abstract public class AbstractStorage {
	public boolean setPermissionForPlayer(UUID id, String name, boolean permission) throws StorageException {
		return setPermissionForPlayer(id, name, permission, true);
	}

	abstract public boolean setPermissionForPlayer(UUID id, String name, boolean permission, boolean update) throws StorageException;

	abstract public Boolean getPermissionForPlayer(UUID id, String name) throws StorageException;
}

File: src/main/kotlin/net/moltendorf/bukkit/linkedwhitelist/storage/AbstractStorage.kt
package net.moltendorf.bukkit.linkedwhitelist.storage

import java.util.UUID

/**
 * Created by moltendorf on 15/07/27.

 * @author moltendorf
 */
abstract class AbstractStorage {
  fun setPermissionForPlayer(id: UUID, name: String, permission: Boolean): Boolean {
    return setPermissionForPlayer(id, name, permission, true)
  }

  @Throws(StorageException::class)
  abstract fun setPermissionForPlayer(id: UUID, name: String, permission: Boolean, update: Boolean): Boolean

  @Throws(StorageException::class)
  abstract fun getPermissionForPlayer(id: UUID, name: String): Boolean?
}
